File: urbanairship-react-native/android/src/main/java/com/urbanairship/reactnative/events/ShowInboxEvent.java
/* Copyright Urban Airship and Contributors */

package com.urbanairship.reactnative.events;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.facebook.react.bridge.Arguments;
import com.facebook.react.bridge.ReadableMap;
import com.facebook.react.bridge.WritableMap;
import com.urbanairship.reactnative.Event;

/**
 * Show inbox event.
 */
public class ShowInboxEvent implements Event {

    private static final String SHOW_INBOX_EVENT = "com.urbanairship.show_inbox";
    private static final String MESSAGE_ID = "messageId";

    private final String messageId;

    /**
     * Default constructor.
     *
     * @param messageId The optional message ID.
     */
    public ShowInboxEvent(@Nullable String messageId) {
        this.messageId = messageId;
    }

    @NonNull
    @Override
    public String getName() {
        return SHOW_INBOX_EVENT;
    }

    @NonNull
    @Override
    public ReadableMap getBody() {
        WritableMap map = Arguments.createMap();
        map.putString(MESSAGE_ID, messageId);
        return map;
    }
    @Override
    public boolean isForeground() {
        return true;
    }
}

File: urbanairship-react-native/android/src/main/java/com/urbanairship/reactnative/events/ShowInboxEvent.kt
/* Copyright Urban Airship and Contributors */

package com.urbanairship.reactnative.events

import com.facebook.react.bridge.Arguments
import com.facebook.react.bridge.ReadableMap
import com.urbanairship.reactnative.Event

/**
 * Show inbox event.
 *
 * @param messageId The optional message ID.
 */
class ShowInboxEvent(private val messageId: String?) : Event {

    override val name = SHOW_INBOX_EVENT

    override val body: ReadableMap = Arguments.createMap().apply {
        putString(MESSAGE_ID, messageId)
    }

    override val isForeground = true

    companion object {
        private const val SHOW_INBOX_EVENT = "com.urbanairship.show_inbox"
        private const val MESSAGE_ID = "messageId"
    }
}